server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -769,296 +640,107 @@ public void start() {
         final Repository repository = repositoriesService.get().repository(remoteStoreRepo);
         assert repository instanceof BlobStoreRepository : "Repository should be instance of BlobStoreRepository";
         blobStoreRepository = (BlobStoreRepository) repository;
-        remoteClusterStateCleanupManager.start();
-        this.remoteRoutingTableService.ifPresent(RemoteRoutingTableService::start);
-    }
-
-    private ClusterMetadataManifest uploadManifest(
-        ClusterState clusterState,
-        List<UploadedIndexMetadata> uploadedIndexMetadata,
-        String previousClusterUUID,
-        UploadedMetadataAttribute uploadedCoordinationMetadata,
-        UploadedMetadataAttribute uploadedSettingsMetadata,
-        UploadedMetadataAttribute uploadedTemplatesMetadata,
-        Map<String, UploadedMetadataAttribute> uploadedCustomMetadataMap,
-        boolean committed
-    ) throws IOException {
-        synchronized (this) {
-            final String manifestFileName = getManifestFileName(
-                clusterState.term(),
-                clusterState.version(),
-                committed,
-                MANIFEST_CURRENT_CODEC_VERSION
+        // this.remoteRoutingTableService.ifPresent(RemoteRoutingTableService::start);
+        String clusterName = ClusterName.CLUSTER_NAME_SETTING.get(settings).value();
+        RemoteWritableEntityStore<Metadata, RemoteGlobalMetadata> globalMetadataBlobStore = new RemoteClusterStateBlobStore<>(
+            getBlobStoreTransferService(),
+            blobStoreRepository,
+            clusterName,
+            threadpool,
+            ThreadPool.Names.GENERIC
```

Comment:
Will prefer a dedicated thread pool

Reply:
Created a new Scaling threadpool for remote_state_read.